src/clients/FtpGateway.cc

```diff
@@ -1210,6 +1208,7 @@ ftpReadWelcome(Ftp::Gateway * ftpState)
         if (nullptr != ftpState->ctrl.message)
             debugs(9, DBG_IMPORTANT, "FTP server is busy: " << ftpState->ctrl.message->key);
 
+        ftpState->scheduleReadControlReply();
```

Comment:
Without this fix Squid gets stuck after getting 120 response. FtpRelay does not have this problem.